pyat/at/collective/Haissinski.py

```diff
@@ -0,0 +1,257 @@
+import numpy
+from at import envelope_parameters, get_mcf
+from at.constants import clight, e_mass, qe
+from scipy.interpolate import interp1d
+import time
+
+class Haissinski(object):
+
+    '''
+    Class to find the longitudinal distribution in the presence 
+    of a short range wakefield.
+
+    This class is a direct implementation of the following paper:
+    "Numerical solution of the Haïssinski equation for the equilibrium 
+    state of  a stored electron beam", R. Warnock, K.Bane, Phys. Rev. Acc.
+    and Beams 21, 124401 (2018)
+
+    The reader is referred to this paper for a description of the methods.
+    The equation number of key formula are written next to the relevant function.
+
+    As input:
+    wake_element is a wake_object that contains a WakeT and WakeZ array. 
+    ring is a ring instance which is needed for machine parameters (sigma_l, sigma_e, etc)
+    
+    m is the number of points in the full distribution that you want
+    kmax is the min and max of the range of the distribution. See equation 27. In units of sigma_z0
+    current is the bunch current.
+    numIters is the number of iterations
+    eps is the convergence criteria.
+
+    the functions solve or solve_steps can be used after initialisation
+
+    Future developments of this class:
+        Adding LR wake or harmonic cavity as done at SOLEIL. Needs to be added WITH this class which is just for short range wake.
+    '''
+
+    def __init__(self, wake_element, ring, m=12, kmax=1, current=1e-4, numIters = 10, eps = 1e-10):
+
+        self.circumference = ring.circumference
+        self.energy = ring.energy
+
+        envelpars = envelope_parameters(ring.radiation_on(copy=True))
+        self.f_s = envelpars.f_s
+        self.nu_s = self.f_s / (clight/self.circumference)
+        self.sigma_e = envelpars.sigma_e * self.energy
+        self.sigma_l = envelpars.sigma_l
```

Comment:
`f_s`, `sigma_e` and `sigma_l` are available in radiation_parameters (analytically computed), which does not need to turn radiation on.